hera_cal/lstbin_simple.py

```diff
@@ -348,36 +358,102 @@ def reduce_lst_bins(
                 days_binned[:, lstbin]
             ) = lst_average(
                 d, n, f, 
-                inpainted=inpf,
                 inpainted_mode=inpainted_mode,
-                mutable=mutable,
-                flag_thresh=flag_thresh,
                 sigma_clip_thresh=sigma_clip_thresh,
                 sigma_clip_min_N=sigma_clip_min_N,
                 flag_below_min_N=flag_below_min_N,
             )
+
+            if get_mad:
+                med[:, lstbin], mad[:, lstbin] = get_lst_mad(d)
         else:
             out_data[:, lstbin] *= np.nan
             out_flags[:, lstbin] = True
             out_std[:, lstbin] *= np.inf
             out_nsamples[:, lstbin] = 0.0
 
+            if get_mad:
+                mad[:, lstbin] *= np.inf
+                med[:, lstbin] *= np.nan
 
-    return out_data, out_flags, out_std, out_nsamples, days_binned
+    out = {
+        "data": out_data,
+        "flags": out_flags,
+        "std": out_std,
+        "nsamples": out_nsamples,
+        "days_binned": days_binned,
+    }
+    if get_mad:
+        out["mad"] = mad
+        out["median"] = med
+
+    return out
 
 def _allocate_dfn(shape: tuple[int], d=0.0, f=0, n=0):
     data = np.full(shape, d, dtype=complex)
     flags = np.full(shape, f, dtype=bool)
     nsamples = np.full(shape, n, dtype=float)
     return data, flags, nsamples
 
-@profile
-def lst_average(
+def get_masked_data(
     data: np.ndarray, nsamples: np.ndarray, flags: np.ndarray,
     inpainted: np.ndarray | None = None,
     inpainted_mode: bool = False,
+) -> np.ma.MaskedArray:
+    if not inpainted_mode:
+        # Act like nothing is inpainted.
+        inpainted = np.zeros(flags.shape, dtype=bool)
+    elif inpainted is None:
+        # Flag if a whole blt is flagged:
+        allf = np.all(flags, axis=2)[:,:, None, :]
+        
+        # Assume everything else that's flagged is inpainted.
+        inpainted = flags.copy() * (~allf)
+
+    flags = flags | np.isnan(data) | np.isinf(data) | (nsamples == 0)
+    data = np.ma.masked_array(
+        data, 
+        mask=(flags & ~inpainted)
+    )
+    return data, flags
+
+
+def get_lst_mad(
```

Comment:
This does both med and mad, so I feel like both should be in the name.

Reply:
Got it. Fixed that up.